IfcLInearZone/IfcGeotechTypicalSection

larswik opened the pull request:
Partly resolves #447

larswik committed:
IfcLInearZone/IfcGeotechTypicalSection

Partly resolves #447

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeotechTypicalSection/DocEntity.xml:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeotechTypicalSection" UniqueId="69b8ea2f-0b5a-4686-86a3-23811f54c68a" BaseDefinition="IfcLinearZone" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="e269468d-3360-41be-8f1d-7feb02283978" DefinedType="IfcGeotechTypicalSectionTypeEnum" AttributeFlags="1">
```
Probably make it mandatory.

anborr replied:
I agree.

larswik replied:
yes!

larswik replied:
Done.

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeotechTypicalSection/DocEntity.xml:
```diff
@@ -0,0 +1,17 @@
+		</DocAttribute>
+	</Attributes>
+	<WhereRules>
+		<DocWhereRule Name="CorrectPredefinedType" UniqueId="434a01d7-f0fb-4b4a-ba95-067dec9d5639">
```
With the above comment in mind we would then also change the rule.

anborr replied:
I agree.

larswik replied:
Yes!

larswik replied:
Done.

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcLinearZone/DocEntity.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcLinearZone" UniqueId="6b26ca28-e2b8-4750-b7ae-ed25437db4ef" BaseDefinition="IfcLinearElement" EntityFlags="32" />
```
Do we make it abstract if we are adding children?

larswik replied:
Depends. Right now we don't have a comprehensive list of children.

larswik replied:
Made it abstract since it is a non-leaf entity. See [f5c1884](https://github.com/bSI-InfraRoom/IFC-Specification/pull/455/commits/f5c18846ce333b199543fd0ce1262fc36c73c33a)

SergejMuhic reviewed (commented):
Just some thoughts for now.

SergejMuhic commented:
Converting to draft for the time being. Still some conversation required.

anborr commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeotechTypicalSection/Documentation.md:
```diff
@@ -0,0 +1 @@
+Interval along the alignment/building structure with similar ground conditions, as part of the GeotechSynthesis model that represents the connection between the ground model and the building. Includes key-properties like expected distribution of ground types (reference to GeotechUnits) and baseline-definition of expected ground conditions and potential hazards, and may also include key-informatio ...
```
I propose to replace 'building' by 'built structure' (2 occurences).

larswik replied:
Agreed.

larswik replied:
Done.

anborr commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeotechTypicalSectionTypeEnum/Documentation.md:
```diff
@@ -0,0 +1 @@
+This container defines the different predefined types of geotech typical sections that can specify an _IfcGeotechTypicalSection_.
```
"geotech typical sections" should become "typical geotechnical sections"

larswik replied:
Agreed.

larswik replied:
Done.

anborr commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcLinearZone/Documentation.md:
```diff
@@ -0,0 +1,3 @@
+A potentially overlapping linear element.
```
Should we say something about the (non-existing) geometry of such a zone?

larswik replied:
Yes, probably. I will draft something. Any help is appreciated.

larswik replied:
Drafted something. Please check!

anborr commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeotechTypicalSectionTypeEnum/DocEnumeration.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeotechTypicalSectionTypeEnum" UniqueId="2b1bd183-94e5-4c80-9682-2ef94c48c7b4">
+	<Constants>
+		<DocConstant xsi:nil="true" href="NOTDEFINED_0hF15nkf79uy7wMtnpxz" />
```
Do we really have only NOTDEFINED and USERDEFINED?

larswik replied:
For now. No requirements for more specific types were specified as of yet.

larswik commented:
I will draft and commit some proposals tomorrow (Tue 9/8) for the above comments.

larswik committed:
Fixed according to comments

- WR => HasObjectType
- PredefinedType => Mandatory
- "Building" => "Built structure"
- "Container" => "Enumeration"
- "geotech typical sections" => "Typical geotechnical sections"
- Wrote a note that geometry is not required and developed the description further

larswik committed:
Made IfcLinearZone abstract

SergejMuhic committed:
Merge branch 'tunnel' into pr/455
